### ISSUE TITLE
Current Freezes

### ISSUE THREAD

**Issue by wrexbe** [space-wizards/space-station-14#8524]
This is a list of things that will probably not get merged right now, while you can still make a PR, you may have to resolve conflicts after the PR that is causing the freeze is merged.

Freezes / PRs that are in progress
* Maps - https://github.com/space-wizards/space-station-14/pull/8301
* UI - https://github.com/space-wizards/space-station-14/pull/7202

**Comment by Zumorica** [comment]
Instruments - Currently refactoring this entirely and moving all of the networking logic to engine. I'm also refactoring most of the engine MIDI code.

**Comment by EmoGarbage404** [comment]
EntityStorage - Undergoing a refactor at #9291

**Comment by ZeroDayDaemon** [comment]
Bartending drinks / reagents / sprites - Undergoing full conversion to /tg/ with the new /tg/ sprites

**Comment by EmoGarbage404** [comment]
Uplink as of #10201

**Comment by metalgearsloth** [comment]
Antags - We haven't actually had a meeting since I proposed it so it's tentatively here.
Reason is people work on a new antag (gameplay for one person occasionally) instead of job content (gameplay actively in every round for several players) so we continue our spiral of having no job content for departments like science.

Fixing an existing antag is okay.

**Comment by Injazz** [comment]
xenosci was merged so consider lifting freeze from research

**Comment by metalgearsloth** [comment]
do_afters as keron is currently refactoring them to be less poopy.

**Comment by juliangiebel** [comment]
Anything related to Machile linking. So using or changing calls to methods of [SignalLinkerSystem](https://github.com/space-wizards/space-station-14/blob/master/Content.Server/MachineLinking/System/SignalLinkerSystem.cs).
I have a refactor PR up that moves linking on top of device networking. #13645

**Comment by EmoGarbage404** [comment]
Gonna ask that people don't touch gamemodes/station events while i rework on gamerule refactor.

**Comment by keronshb** [comment]
Restoring the antag freeze.

**Comment by deltanedas** [comment]
does borging count as species i wanna try it

**Comment by metalgearsloth** [comment]
> does borging count as species i wanna try it

We're discussing freezes more in discord but no. Borgs need to use body system though.

**Comment by deltanedas** [comment]
yeah ill be basing it off current body rework code, seems like brain will be fun fun fun

**Comment by EmoGarbage404** [comment]
Requesting that all prs adding new technologies and modifications to the RND system be either merged soon or postponed while i begin work on the [research rework.](https://hackmd.io/@ss14-design/Sk1VIWjni)

**Comment by AjexRose** [comment]
Is technology unfrozen since Emo got their tech research rework finished?

**Comment by EmoGarbage404** [comment]
> Is technology unfrozen since Emo got their tech research rework finished? 

Yea

**Comment by moonheart08** [comment]
Don't mess with stations or gamemodes at the moment please, reworking how stations work under the hood. This means don't modify the prototypes, mapping is fine and won't interfere with me.

**Comment by keronshb** [comment]
Added Wizard since it still would require Mind Entities and the highest scrutiny.

**Comment by Boaz1111** [comment]
Regarding the freeze on weapons, would a PR making certain weapons(musket, etc.) obtainable via a research be frozen?

**Comment by keronshb** [comment]
Added item balance because we're not at a place where we have consistency internally.

**Comment by keronshb** [comment]
Readded Wizard and Magic to freezes

**Comment by VasilisThePikachu** [comment]
I fat-fingered remove, don't mind me

**Comment by he1acdvv** [comment]
![better-call-saul-faint](https://github.com/user-attachments/assets/d0afb9d9-ebb3-4898-86af-5c1d718ead95)

**Comment by Everturning** [comment]
@wrexbe loadouts have been in for a while now will you edit the roundstart clothing thing

**Comment by VasilisThePikachu** [comment]
> @wrexbe loadouts have been in for a while now will you edit the roundstart clothing thing

I can do it. Maints can edit posts.

**Comment by Golinth** [comment]
> Variants will not be accepted until we support displacement masks

Should clothing variants for mobs be removed from the freeze, now that we support displacement maps?

**Comment by MilonPL** [comment]
Removed Wizard from the list. This includes stuff related to the wizard antag itself like the shuttle, spawn rates, etc. Spells are still a hard no.
Added a freeze on cartridges and PDA code.

**Comment by keronshb** [comment]
Added Actions to the freeze. This includes anything like:
- Action ECS
- Yaml adjustments to any existing actions (additions/changes/removals)
- Anything related to actions

This is so #27422 can be merged to vulture without conflicts and so that we have the maximum amount of time testing from the next stable merge. It's a huge PR and needs every little bit of testing we can muster.

**Comment by Chromaphasia** [comment]
The action rework has been implemented and merged but this page still lists actions as frozen

**Comment by sowelipililimute** [comment]
`StyleNano` is on this list now, use the new sheetlets instead.

**Comment by FairlySadPanda** [comment]
Would like to propose polymorph and all temporary entity storage to be added to this list. This in turn would mean Ling is formally frozen until the above is done.

The codebase does not currently handle being able to stash and retrieve in-play entities in a consistent way, leading to stupid, hard-to-debug race conditions, amongst other things.

Being able to push and pop entities to a repository outside the game's playspace is a core tech feature and needs to be done via core tech, not via various methods taken within the codebase.

Likewise this applies to any PR that needs to work on components with semantically-weak references to other entities on them. This has been blockaded by inaction on the engine side for months. We can't do a sensible entity storage system that allows for referenced entities stored inside entities in the repository to be deleted without some solution that allows for this.

**Comment by slarticodefast** [comment]
I added **Power Cells & Batteries** to the list of freezes.

**Comment by keronshb** [comment]
I added Polymorph per FSP

**Comment by slarticodefast** [comment]
@keronshb I don't think adding a polymorph freeze makes sense in this form. We discussed it a little on discord [here](https://discord.com/channels/310555209753690112/770682801607278632/1437407736013914192)
It should have some clear requirements on what is needed rather than just outright blocking off an entire part of the codebase without anyone actively working on it. And any changes to this would likely not need changes to the polymorph API, so I don't see why a content freeze for that is needed in the first place.

I'll remove it from the list for now until we have a consensus on that.

**Comment by EthanQix** [comment]
Is there an issue about what PRs contributors should prioritize, kind of a TODO list to complement this NOTTODO list ?

**Comment by slarticodefast** [comment]
> Is there an issue about what PRs contributors should prioritize, kind of a TODO list to complement this NOTTODO list ?

We have a work in progress roadmap here if you want to have a look https://hedgedoc.spacestation14.com/s/5wpBdb7Oq#
In the near future we will create issues with exact requirements for each of the points in there and collect them in a github project so that contributors will have a better overview on what they can work on.

**Comment by slarticodefast** [comment]
I removed power cells and batteries from the list since the refactor/prediction PR has been merged.

**Comment by slarticodefast** [comment]
Added a freeze for Markings/Species: Janet is currently working on a refactor for these for future plans with surgery and other medical systems. This freeze lasts until the efforts for refactoring HumanoidAppearance are done. The scope is Content.*.Humanoid, Content.*.Body, and components/prototypes defined with their types.

**Comment by Princess-Cheeseballs** [comment]
Added a freeze for Mapping. Emisse is currently working on mapping guidelines and principles for maps to be held to. The scope is all maps including stations, salvage wrecks, and shuttles.

**Comment by Qwazori** [comment]
Does the microwave recipe ban also apply to everything except food? Are medical recipes also frozen?

**Comment by VasilisThePikachu** [comment]
> Does the microwave recipe ban also apply to everything except food? Are medical recipes also frozen?

yes it includes those too, sorry

**Comment by slarticodefast** [comment]
Silicons were temporarily added to the list of freezes.

> Silicons @ScarKy0 is currently refactoring the silicon law system, due to the nature of the refactor relevant code has been frozen to avoid merge conflicts. Important bugfixes and similiar can bypass the freeze, just ask the silicon workgroup first. The scope is Content.*/Silicons and the relevant yaml files. This freeze will last until https://github.com/space-wizards/space-station-14/pull/41659 and https://github.com/space-wizards/space-station-14/pull/42995 are merged.

**Comment by metalgearsloth** [comment]
NPCs and pathfinding.
- Pathfinder so we can get regions back and work on a space pathfinder / make obstacle clearing cleaner.
- NPCs because it's too scrunkly to use and I now dislike HTN. I may or may not approve operator PRs.

If you want to work on either please ask me first with some sort of long-term strategy or else I'm liable to close your PR.

**Comment by chromiumboy** [comment]
I have removed the freeze on disposals related PRs from the list

**Comment by SlamBamActionman** [comment]
Structures & Tiles Sprites have been added to the list of freezes.  
> * **Structure & Tile Sprites** @AftrLite is making major changes to our base sprites and to avoid merge conflicts we're asking that changes to the Texture folders for `Structures/Walls`, `Structures/Windows`, `Structures/Wallmounts/Lighting`, `Structures/Doors` and `Tiles` are checked with him (Kazne on Discord).
